rskj-core/src/test/java/co/rsk/peg/PendingFederationTest.java

```diff
@@ -129,86 +134,129 @@ public void testToString() {
-                    mock(ForBlock.class)
-                )
-        );
+        Assert.assertEquals(expectedFederation, builtFederation);
     }
 
     @Test
     public void buildFederation_ok_9_members_before_RSKIP_199_activation() {
-        PendingFederation otherPendingFederation = new PendingFederation(
-            FederationTestUtils.getFederationMembersFromPks(
-                100, 200, 300, 400, 500, 600, 700, 800, 900
-        ));
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP199)).thenReturn(false);
 
-        Federation expectedFederation = new Federation(
-                FederationTestUtils.getFederationMembersFromPks(
-                    100, 200, 300, 400, 500, 600, 700, 800, 900
-                ),
-                Instant.ofEpochMilli(1234L),
+        Integer[] privateKeys = new Integer[]{100, 200, 300, 400, 500, 600, 700, 800, 900};
+        Instant creationTime = Instant.ofEpochMilli(1234L);
+
+        PendingFederation otherPendingFederation = new PendingFederation(
+            FederationTestUtils.getFederationMembersFromPks(privateKeys)
+        );
+        Federation builtFederation = otherPendingFederation.buildFederation(
+            creationTime,
             0L,
-                NetworkParameters.fromID(NetworkParameters.ID_REGTEST)
+            bridgeConstants,
+            activations
         );
 
-        Assert.assertEquals(
-                expectedFederation,
-                otherPendingFederation.buildFederation(
-                        Instant.ofEpochMilli(1234L),
-                        0L,
-                        bridgeConstants,
-                    mock(ForBlock.class)
-                )
+        Federation expectedFederation = new Federation(
+            FederationTestUtils.getFederationMembersFromPks(privateKeys),
+            creationTime,
+        0L,
+            NetworkParameters.fromID(NetworkParameters.ID_REGTEST)
         );
+
+        Assert.assertEquals(expectedFederation, builtFederation);
+    }
+
+    @Test
+    public void buildFederation_erp_ok_after_RSKIP_201_activation_before_RSKIP284_testnet() {
+        testBuildErpFederation(false, NetworkParameters.ID_TESTNET);
+    }
+
+    @Test
+    public void buildFederation_erp_ok_after_RSKIP_201_activation_before_RSKIP284_mainnet() {
+        testBuildErpFederation(false, NetworkParameters.ID_MAINNET);
     }
 
     @Test
-    public void buildFederation_erp_ok_after_RSKIP_201_activation() {
+    public void buildFederation_erp_ok_after_RSKIP_201_activation_after_RSKIP284_testnet() {
+        testBuildErpFederation(true, NetworkParameters.ID_TESTNET);
+    }
+
+    @Test
+    public void buildFederation_erp_ok_after_RSKIP_201_activation_after_RSKIP284_mainnet() {
+        testBuildErpFederation(true, NetworkParameters.ID_MAINNET);
+    }
+
+    private void testBuildErpFederation(boolean isRskip284Active, String networkId) {
```

Comment:
would it make sense to give this method a more meaningful name?
```suggestion
    private void testBuildErpFederationConsideringRskip284Activation(boolean isRskip284Active, String networkId) {
```

Reply:
done